src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -29,70 +30,80 @@ protected AddressBook getInitialData() {
+        PersonCardHandle alicePersonCard = personListPanel.getPersonCardHandle(td.alice);
 
+        personListPanel.rightClickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =
+                personListPanel.clickOnContextMenu(PersonListPanelHandle.ContextMenuChoice.EDIT);
+        assertTrue(editPersonDialog.isShowingPerson(td.alice));
         //Prepare new values for Alice
         Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
                 .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
                 .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
-
-        //Get a reference to the card displaying Alice's details
-        PersonCardHandle alicePersonCard = personListPanel.getPersonCardHandle(td.alice);
-
-        //Edit Alice to change to new values
-        personListPanel.clickOnPerson(td.alice);
-        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
-        editPersonDialog.enterNewValues(newAlice);
-        editPersonDialog.pressEnter();
+        editPersonDialog.enterNewValues(newAlice).pressEnter();
+        assertMatching(alicePersonCard, newAlice);
 
         //Confirm pending state correctness
         assertTrue(alicePersonCard.isShowingGracePeriod("Editing"));
 
         //Confirm the right card is selected after the edit
-        assertEquals(alicePersonCard, newAlice);
         assertTrue(personListPanel.isSelected(newAlice));
 
-
         //Confirm right values are displayed after grace period is over
         sleepForGracePeriod();
-        assertEquals(alicePersonCard, newAlice);
+        assertMatching(alicePersonCard, newAlice);
 
+        //Confirm cancel operation does not cancel the edit after the grace period.
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
         //Confirm the underlying person object has the right values
-        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
-
+        assertMatching(alicePersonCard, newAlice);
 
         //Confirm again after the next sync
         sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
         //Confirm other cards are unaffected.
-        personListPanel.clickOnListView();
         assertTrue(personListPanel.isListMatching(1, td.benson, td.charlie, td.dan, td.elizabeth));
 
         //Confirm status bar is updated correctly
-        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] completed successfully.");
+        assertEquals(HeaderStatusBarHandle.formatSuccessMessage(HeaderStatusBarHandle.Type.EDIT, td.alice.fullName(),
+                     Optional.of(newAlice.fullName())), statusBar.getText());
```

Comment:
If without the type. We need to have 3 different type of similar methods.
getEditCancelledMessage(), getAddCancelledMessage(), getDeleteCancelledmessage.

And "get.." seems to suggesting it is getting something from the statusBar. 

Messages may need two different names. For e.g. for this case [ Alice Brown -> Alicia Brownstone ].

All these cases complicate the convenience method. I guess showing the plain text will be more appropriate?